doc/smart-switch/BFD/SmartSwitchDpuLivenessUsingBfd.md

```diff
@@ -0,0 +1,265 @@
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    "mac_address":{{mac_address1},{mac_address2},...} (OPTIONAL)  
+    "vni": {{vni1},{vni2},...} (OPTIONAL) 
+    "weight": {{w1},{w2},...} (OPTIONAL) 
+    “profile”: {{profile_name}} (OPTIONAL) 
+    "rx_monitor_timer": {time in milliseconds}
+    "tx_monitor_timer": {time in milliseconds}
+```
+
+## 2.2 SmartSwitch NPU-DPU BFD sessions
+
+In a Smartswitch deployment, BFD sessions are created between NPU and
+DPUs. Each T1 cluster has 8 T1s and 8 DPUs in each T1. To detect the
+liveness of each DPU in the cluster, NPU creates Active BFD sessions to
+all 64 DPUs and each DPU creates passive BFD sessions to all NPUs in the
+cluster. When BFD keepalives are not detected for certain interval, BFD
+sessions are marked as down and corresponding action is performed.
+Multihop BFD sessions are created between NPU and a remote DPU.
+
+<img src="images/NPU_DPU_BFD_sessions.png">
+
+## 2.3 BFD Active-Passive mode
+
+HA manager in NPU creates VNET_ROUTE_TUNNEL_TABLE which will trigger bfdOrch to create BFD Active session with local and peer information. NPU supports BFD hardware offload, so hardware starts
+generating BFD packets to the peer. HA manager also creates another
+BFD_SESSION table in DPU with local DPU PA and peer NPU PAs. BfdOrch checks for SAI object platform capability for h/w offload. In DPU, if h/w offload is not supported, FRR is configured to trigger BFD sessions. BfdOrch creates SOFTWARE_BFD_SESSION_TABLE entry in STATE_DB with 'passive' mode and BgpCfgd checks for the entries in the STATE_DB and configures BFD passive sessions in
+FRR stack with local and peer information.
+
+SOFTWARE_BFD_SESSION_TABLE entry in STATE_DB has all the key fields and values from BFD_SESSION_TABLE.
+
+Trigger for HA manager to create BFD sessions will be detailed in HA HLD.
+
+<img src="images/Dash_BFD_FRR.png">
+
+BFD passive sessions in DPU waits for BFD control packets from NPU's
+active session. When FRR stack receives BFD packets from active sessions
+in NPU, it starts responding. After the initial handshake, BFD sessions
+are up. When active session doesn't receive control packets for the
+configured BFD timeout period, session is set to DOWN and corresponding
+action is taken.
+
+<img src="images/NPU_DPU_BFD_Exchange.png">
+
+## 2.4 DPU FRR Changes
+
+By default BFD Daemon is not enabled in FRR. FRR supervisor config file
+has to be updated to start BFDD during FRR initialization.
+
+To configure BFD passive sessions, a profile with 'passive' mode is
+created and used for all sessions created in FRR. BgpCfgd configures FRR
+stack using vtysh.
+
+```
+
+bfd
+
+profile passive
+
+passive-mode
+
+receive-interval 100
+
+transmit-interval 100
+
+detect multiplier 3
+
+exit
+
+!
+
+peer \<Local/Remote NPU IP\> local-address \<DPU PA\>
+
+profile passive
+
+exit
+```
+

```

Comment:
we need a notification channel from DPU to NPU to give hamgrd the BFD status. this will be used by SDN controller as one of the DPU health signals (e.g. all BFD session is dead is probably not a good sign)

Reply:
section added with reference to HA doc.